### ISSUE TITLE
Using Tower model from Emberjs -- url is not defined 

### ISSUE THREAD

**Issue by avaranovich** [tower-archive/tower#194]
I am trying to create a resource from Ember view (by handling submitLogin action)

``` coffeescript
      MyApp.SignUpView = Ember.View.extend
       templateName: 'signup'

       submitLogin: (event) ->
        login = @getPath('email.value')
        password = @getPath('password.value')
        console.log "Login: " + login + " Password: " + password
        user = new App.User
        user.email = login
        user.password = password
        console.log Tower.urlFor(user)
        console.log user
        App.User.create user
```

The code above works correctly up to the last line: App.User.create user
which generates following stacktrace (I list relevant part of it)
Uncaught ReferenceError: url is not defined tower.js:2944
Tower.Store.Ajax.Ajax.createRequest tower.js:2944
Tower.Store.Ajax.Ajax.request tower.js:2825
Tower.Store.Ajax.Ajax.queue tower.js:2838
Tower.Store.Ajax.Ajax.queue tower.js:2873
Tower.Store.Ajax.Ajax.createRequest tower.js:2941
Tower.Store.Ajax.Ajax.create tower.js:2898
Tower.Store.Memory.Persistence.create tower.js:2600
Tower.Store.Ajax.Ajax.create tower.js:2894
Tower.Model.Criteria.Criteria._create tower.js:3659
Tower.Model.Criteria.Criteria.create tower.js:3621
Tower.Model.Scope.Scope.create tower.js:3250
Tower.Model.Persistence.InstanceMethods._create tower.js:5755
Tower.Support.Callbacks.Chain.Chain.run tower.js:902
_.extend.series.iterate tower.js:1796
Tower.Support.Callback.Callback.run tower.js:975
Tower.Support.Callbacks.Chain.Chain.run.runner

Please note I am using MyApp.\* for Emberjs-related views/routers/controllers
and App.\* for towerjs

**Comment by avaranovich** [comment]
I looked at tower.js

``` javascript
 Ajax.prototype.createRequest = function(records, options) {
       .....
       json = this.toJSON(records);
       Tower.urlFor(records.constructor); // should this be url = Tower.urlFor(records.constructor); ??
       return this.queue(function() {
         var params;
         params = {
           url: url, //otherwise it is not clear where this url is assigned
           type: "POST",
           data: json
         };
        return _this.ajax(options, params).success(_this.createSuccess(records)).error(_this.createFailure(records));
      });
```

**Comment by edubkendo** [comment]
Looks like the function you are referencing will be handled differently in the next version of tower (0.4.1) which will be released soon. In it, url is assigned as you say, "url = ..." . Did you try changing it in the file to see if your code would run?
By the way, when you quote code here on github, if you put  three backticks followed by "javascript " (or coffeescript, or ruby, or whatever) on the line above, and three backticks on the line below, you'll get pretty formatting with syntax highlighting which makes it a lot easier to read.

**Comment by avaranovich** [comment]
If I change the url, then I see the following data being send to the server as a POST request:

``` json
{"user":{"id":"0","createdAt":"2012-08-09T13:33:38.659Z","updatedAt":"2012-08-09T13:33:38.659Z"}}
```

And I get another error on completion:

Uncaught TypeError: Cannot call method 'conditions' of undefined tower.js:2346
Tower.Store.Memory.Finders.find tower.js:2346
Tower.Store.Ajax.Ajax.createSuccess tower.js:2957
f.Callbacks.n jquery.min.js:2
f.Callbacks.o.fireWith jquery.min.js:2
w jquery.min.js:4
f.support.ajax.f.ajaxTransport.send.d

So basically the User properties (email and password) are not sent to the server. Any ideas?

the same happens if I use 

``` javascript
user.save()
```

**Comment by edubkendo** [comment]
Ah, toying around with the browser console a bit, I think I figured it out. You have to use the proper accessors to change those properties. So:

``` coffeescript
user.set('email', login)
user.set('password', password)
user.save()
```

Similarly, to retrieve object properties, use the 'get' accessor.

``` coffeescript
email = user.get('email')
password = user.get('password')
```

Hopefully this will work. I'm running the development branch, with Ember.objects built in, so things may still be different from your set up but I think this should do it.

**Comment by avaranovich** [comment]
@edubkendo That was the trick indeed. I am actually quite puzzled now, since I can create only one user. Perhaps this is related to the other error I reported above (Uncaught TypeError: Cannot call method 'conditions' of undefined tower.js:2346) I still see in the console that the data comes to the controller:
 Processing by UsersController#create as JSON
  Parameters:
....
But the 2nd user is not saved to the database.

**Comment by avaranovich** [comment]
The issues seems to be that _id was not generated, and the entity in fact saved with _id = 0.
